### PR TITLE
CI: Move autoformat to master

### PR DESCRIPTION
**Pull request by bnchrch**
## Proposal
Today, Autoformatting occurs at the PR level.

This is difficult for a few reasons
1. Auto commit can interfere with required checks being reported 
2. Auto commit can cause tests to run twice (expensive)
3. A culture that uses `approve-and-merge` leads to unformatted code reaching master, then being formatted in a subsequent PR

We should 
1. Only autoformat + commit on master (This PR)
2. Have ide extensions and/or commit hooks support (Later changes)

Original credit: @postamar @alafanechere 
Document with discussion: https://docs.google.com/document/d/1X8KVmomXT5EwiNuwyvO5oQGMTMrZbt1nyBxWsSpDEo8/edit?usp=drive_web&ouid=100797160259845229564

### CODE REVIEW

**Comment by aaronsteers** [comment]
I think we are iterating generally in the right direction here.

However, autocommitting to master can be controversial.

I can think of two steps that might help us out:
1. Make autocommits on PRs an opt-in behavior via slash command.
2. Instead of autocommitting to master, we can auto-generate a PR with those changes.

**Comment by postamar** [comment]
What are the arguments against autoformatting master that would make doing so worse (in some respects) than the status quo? Genuinely curious.

Otherwise, LGTM. Since we effectively can't (yet) exert control on what gets merged, let's at least maintain a minimum of sanity in the master branch.

**Comment by bnchrch** [comment]
Currently the way I see the unhappy path **today** is
1. Dev makes a PR
2. Cant or wont wait for CI to complete
3. Uses `/approve-and-merge`
4. Master is now unformatted
5. Downstream PRs will now have unexpected changes after CI runs.

And the resolution is you either:
1. Have to dirty your PR and use those unrelated changes
2. Submit a format only PR.

There are a few possible things we can do that are more involved (new ci, removing approve and merge, PR formats)

But I dont think those are achievable today.

Where as updating format to **also** occur on the master branch can fix our issue today with minimal impact

cc: @postamar @aaronsteers

File: .github/workflows/format.yml
Comment (girarda) [inline .github/workflows/format.yml:61]: nit: the comment should be updated

File: .github/workflows/format.yml
Comment (girarda) [inline .github/workflows/format.yml:63]: should the autoformat only run on master or should it run on PRs too?

**Review by girarda — APPROVED** [review]
approving because I'm onboard with the plan. I haven't tested the updated workflow

**Comment by aaronsteers** [comment]
@bnchrch - This seems like the right direction to me.

In terms of a lovable end state, I think these two would be nice "next steps" from what you have here.

1. Instead of auto-committing to master, we can commit those changes to a new branch and auto-generate a PR.
2. Something like our proposed `/airbyte-ci format` slash command would remove friction on PRs, since it can be executed by the author or by a review, and since it doesn't require running onything on the local machine.

**Comment by bnchrch** [comment]
closing as this should become a non-issue once approve-and-merge is restricted to only be run by eng managers, and after required checks have passed

**Pull request closed without merging**
